internal/storage/token.go

```diff
@@ -21,11 +28,115 @@ type TokenStore interface {
 	Delete(name string) error
 }
 
-func NewTokenStore(name string) TokenStore {
-	switch strings.ToLower(name) {
-	case "keychain":
-		return KeychainTokenStore{}
-	default:
-		panic("storage: unknown TokenStore")
+const keychainSvcName = "zed tokens"
+
+type KeychainTokenStore struct{}
+
+var _ TokenStore = KeychainTokenStore{}
+
+func openKeyring() (keyring.Keyring, error) {
+	path, err := localConfigPath()
+	if err != nil {
+		return nil, err
 	}
+
+	return keyring.Open(keyring.Config{
+		ServiceName: keychainSvcName,
+		FileDir:     filepath.Join(path, "keyring"),
+		FilePasswordFunc: func(prompt string) (string, error) {
+			if password, ok := os.LookupEnv("ZED_KEYRING_PASSWORD"); ok {
+				return password, nil
+			}
+
+			fmt.Fprintf(os.Stderr, "%s: ", prompt)
+			b, err := term.ReadPassword(int(os.Stdin.Fd()))
+			if err != nil {
+				return "", err
+			}
+			fmt.Println()
+			return string(b), nil
+		},
+	})
+}
+
+func (ks KeychainTokenStore) List(redactTokens bool) ([]Token, error) {
```

Comment:
Change to an enum and set the "default" value (i.e. `0`) to be redacted

Reply:
Using an enum is actually shitty for booleans when you can just create a new type from the bool.

I tried doing so, but it doesn't improve the readability of the API or call-sites because we're basically passing the flag value directly into this function (e.g. `storage.DefaultTokenStore.List(cobrautil.MustGetBool("reveal-tokens"))`)

Reply:
It doesn't help readability in that case, but it helps safety a bit still, especially if we end up adding another boolean parameter. I tend to still prefer it

Reply:
If this wasn't an internal API, then I might be swayed simply from the backwards-compatibility argument, but I think this is fine for now. I actually over-engineered the storage library originally and have been simplifying it.